File: Core/warn/systemwarningmessage.cpp
#include "systemwarningmessage.h"

SystemWarningMessage::SystemWarningMessage(QString deviceId, QString msg, QObject *parent) : MessageBase(parent), m_deviceId(deviceId), m_msg(msg)
{
}

SystemWarningMessage::SystemWarningMessage(ClientDevice *device, QString msg, QObject *parent) : MessageBase (parent), m_deviceId(device->id()), m_msg(msg) {
}


MessageBase::MESSAGE_TYPE SystemWarningMessage::getMessageType() {
    return MESSAGE_TYPE_SYSTEM_WARNING;
}

QString SystemWarningMessage::deviceId() {
    return m_deviceId;
}

QString SystemWarningMessage::msg() {
    return m_msg;
}

QString SystemWarningMessage::getFirstLevelId() {
    return m_deviceId;
}

QString SystemWarningMessage::getSecondLevelId() {
    return "";
}

File: AndroidApp/OshCore/src/main/java/com/osh/warn/SystemWarningMessage.java
package com.osh.warn;

import com.osh.communication.MessageBase;

public class SystemWarningMessage extends MessageBase {
	
	private String deviceId;
	private String msg;

	public SystemWarningMessage(String deviceId, String msg) {
		this.deviceId = deviceId;
		this.msg = msg;
	}

	@Override
	public MESSAGE_TYPE getMessageType() {
		return MESSAGE_TYPE.MESSAGE_TYPE_SYSTEM_WARNING;
	}

	@Override
	public String getFirstLevelId() {
		return deviceId;
	}

	@Override
	public String getSecondLevelId() {
		return "";
	}

	public String getDeviceId() {
		return deviceId;
	}

	public void setDeviceId(String deviceId) {
		this.deviceId = deviceId;
	}

	public String getMsg() {
		return msg;
	}

	public void setMsg(String msg) {
		this.msg = msg;
	}
	
}
